src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -424,9 +424,14 @@ protected function getEnv($name)
     }
 
     /**
+     * @param string|false $registry
+     * @param string|bool  $load
+     *
+     * @return mixed
+     *
      * @internal
      */
-    final protected function getService($registry, $id, $method, $load)
+    final protected function getService($registry, string $id, ?string $method, $load)
```

Comment:
On a different note, would it make sense to `null` instead of `false`?

Reply:
It probably would.

Reply:
I've just checked this. If we wanted to have `?string $registry` here, we would need to change `PhpDumper` because that method is used in compiled container code. I could dig further into it, but for now I'd say that this change would be out of scope for this PR.